app/controllers/posts_controller.rb

```diff
@@ -7,4 +8,21 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(params[:id])
+    @post.title = params[:title]
+    @post.text = params[:text]
+    @post.author_id = current_user.id
```

Comment:
- [ ] since you have a `current_user` method, you should use it to create a new post. The code should look like this:
`@comment = current_user.posts.new(post_params)`. This way you don't need to pass the user id in line 20.